From c63652f6ab25b6bbd69000b0612db7be2b1a7690 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Fri, 13 Mar 2020 20:39:10 +0100
Subject: [PATCH 22/36] docking: Ignore key-repeat

Use same keybinding flags as upstream to avoid an icon keybinding to be
triggered multiple times.
---
 docking.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/docking.js b/docking.js
index 8b338f5..1d249c8 100644
--- a/docking.js
+++ b/docking.js
@@ -1341,7 +1341,7 @@ var KeyboardShortcuts = class DashToDock_KeyboardShortcuts {
             for (let i = 0; i < DashToDock_KeyboardShortcuts_NUM_HOTKEYS; i++) {
                 let appNum = i;
                 Main.wm.addKeybinding(key + (i + 1), DockManager.settings,
-                                      Meta.KeyBindingFlags.NONE,
+                                      Meta.KeyBindingFlags.IGNORE_AUTOREPEAT,
                                       Shell.ActionMode.NORMAL | Shell.ActionMode.OVERVIEW,
                                       () => {
                                           DockManager.getDefault().mainDock._activateApp(appNum);
@@ -1402,7 +1402,7 @@ var KeyboardShortcuts = class DashToDock_KeyboardShortcuts {
     _enableExtraShortcut() {
         if (!this._shortcutIsSet) {
             Main.wm.addKeybinding('shortcut', DockManager.settings,
-                                  Meta.KeyBindingFlags.NONE,
+                                  Meta.KeyBindingFlags.IGNORE_AUTOREPEAT,
                                   Shell.ActionMode.NORMAL | Shell.ActionMode.OVERVIEW,
                                   this._showOverlay.bind(this));
             this._shortcutIsSet = true;
-- 
2.25.1

